warp/aggregator/signature_job.go

```diff
@@ -0,0 +1,59 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package aggregator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+var errInvalidSignature = errors.New("invalid signature")
+
+// SignatureBackend defines the minimum network interface to perform signature aggregation
+type SignatureBackend interface {
+	// FetchWarpSignature attempts to fetch a BLS Signature from [nodeID] for [unsignedWarpMessage]
+	FetchWarpSignature(ctx context.Context, nodeID ids.NodeID, unsignedWarpMessage *avalancheWarp.UnsignedMessage) (*bls.Signature, error)
+}
+
+type signatureJob struct {
+	backend SignatureBackend
+	msg     *avalancheWarp.UnsignedMessage
+
+	nodeID    ids.NodeID
+	publicKey *bls.PublicKey
+	weight    uint64
+}
+
+func (s *signatureJob) String() string {
+	return fmt.Sprintf("(NodeID: %s, UnsignedMsgID: %s)", s.nodeID, s.msg.ID())
+}
+
+func newSignatureJob(backend SignatureBackend, validator *avalancheWarp.Validator, msg *avalancheWarp.UnsignedMessage) *signatureJob {
+	return &signatureJob{
+		backend:   backend,
+		msg:       msg,
+		nodeID:    validator.NodeIDs[0], // XXX: should we attempt to fetch from all nodeIDs and use the first valid response?
```

Comment:
When is it possible for validator to have more than 1 node ID?

Reply:
`avalancheWarp.Validator` is different from the notion of a validator on the P-Chain. In this context, a validator is a one to many relationship between a registered BLS Public Key and any nodeIDs that use that public key.

The question in this comment is, if there are multiple nodeIDs with the same BLS Public Key, should we try to fetch a signature from more than one of them and take the first that we get? Right now, we just try to grab it from the first nodeID.